### PR TITLE
refactor(listener): support dynamic query parameters

### LINKED ISSUE #1602: Rubrix listeners
A first implementation version that allows clients to react to dataset changes.

These listeners can be the base for other integration like alerts, active learning loops, model monitoring, and other MLOPs actions.

### PR DESCRIPTION
**Pull request by frascuchon**

### CODE REVIEW

File: tests/listeners/test_listener.py [hunk 1]
```diff
@@ -32,13 +35,23 @@ class TestListener:
...
         def action(self, records: List[Record], ctx: RBListenerContext):
             try:
+                assert ctx.dataset == dataset
+                if ctx.query_params:
+                    assert ctx.query == query.format(**ctx.query_params)
+                    if self.executed:
+                        assert ctx.query_params != query_params
+                    else:
+                        assert ctx.query_params == query_params
+                    ctx.query_params["params"] += 1
```
Comment (frascuchon) [inline tests/listeners/test_listener.py]: Maybe you can just check the query after changing the query params:

```suggestion
                    ctx.query_params["params"] += 1
                    assert ctx.query_params != query_params
                    assert ctx.query == query.format(**ctx.query_params)
```

**Review by dcfidalgo — DISMISSED** [review]

File: tests/listeners/test_listener.py (hunk 1 above)
Comment (dcfidalgo) [reply tests/listeners/test_listener.py]: hm, here the idea was to test if changes to the `ctx.query_params` persist to the next execution of the action. I think i will simply remove the else clause.

Comment (frascuchon) [reply tests/listeners/test_listener.py]: 👍

File: tests/listeners/test_listener.py [hunk 2]
```diff
@@ -32,13 +35,23 @@ class TestListener:
...
         )
         def action(self, records: List[Record], ctx: RBListenerContext):
             try:
+                assert ctx.dataset == dataset
+                if ctx.query_params:
+                    assert ctx.query == query.format(**ctx.query_params)
+                    if self.executed:
+                        assert ctx.query_params != query_params
+                    else:
+                        assert ctx.query_params == query_params
```
Comment (dcfidalgo) [inline tests/listeners/test_listener.py]:
```suggestion
```

**Review by dcfidalgo — APPROVED** [review]

**Pull request merged**